frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -1218,7 +1218,7 @@ const IR::Node* TypeInference::postorder(IR::StructField* field) {
 const IR::Node* TypeInference::postorder(IR::Type_Header* type) {
     auto canon = setTypeType(type);
     auto validator = [] (const IR::Type* t)
-            { return t->is<IR::Type_Bits>() || t->is<IR::Type_Varbits>(); };
+            { return t->is<IR::Type_Bits>() || t->is<IR::Type_Varbits>() || t->is<IR::Type_Struct>(); };
```

Comment:
Please add a comment for this last case labeling it as experimental.

Reply:
Done.